stdlib/src/builtin/bin.mojo

```diff
@@ -0,0 +1,80 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+"""Implements the `bin()` function
+
+These are Mojo built-ins, so you don't need to import them.
+"""
+
+
+@always_inline("nodebug")
+fn bin(b: Scalar[DType.bool]) -> String:
+    """Returns the binary representation of a scalar bool.
+
+    Args:
+        b: A scalar bool value.
+
+    Returns:
+        The binary string representation of b.
+    """
+    return bin(int(b))
+
+
+fn bin[type: DType](num: Scalar[type]) -> String:
+    """Return the binary string representation an integral value.
+
+    ```mojo
+    print(bin(123))
+    print(bin(-123))
+    ```
+    ```plaintext
+    '0b1111011'
+    '-0b1111011'
+    ```
+
+    Parameters:
+        type: The data type of the integral scalar.
+
+    Args:
+        num: An integral scalar value.
+
+    Returns:
+        The binary string representation of num.
+    """
+    constrained[type.is_integral(), "Expected integral value"]()
+    alias BIN_PREFIX = "0b"
+
+    if num == 0:
+        return BIN_PREFIX + "0"
+
+    var result = String()
```

Comment:
**Suggestion** You could pre-allocate the string size here if you'd like to avoid some allocations by pre-computing the size.

Reply:
Unless you had something else in mind I'm not sure we can at the moment due to #2194 since you can't currently call `bitwidth()`  for `index` size integers.